### PR TITLE
TS StrictMode for various components

### PR DESCRIPTION
**Pull request by yihuiliao**
Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-aria/color/src/useColorArea.ts [hunk 1]
```diff
@@ -188,7 +188,7 @@ export function useColorArea(props: AriaColorAreaOptions, state: ColorAreaState)
   });
 
   let currentPointer = useRef<number | null | undefined>(undefined);
-  let isOnColorArea = useRef<boolean>(false);
+  let isOnColorArea = useRef<boolean | undefined>(false);
```
Comment (snowystinger) [inline packages/@react-aria/color/src/useColorArea.ts:191]: the instance where we set this to undefined should be set to false instead. I think the type was correct as it was

File: packages/@react-aria/color/src/useColorWheel.ts [hunk 2]
```diff
@@ -116,7 +116,7 @@ export function useColorWheel(props: AriaColorWheelOptions, state: ColorWheelSta
   let {moveProps: movePropsThumb} = useMove(moveHandler);
 
   let currentPointer = useRef<number | null | undefined>(undefined);
-  let isOnTrack = useRef<boolean>(false);
+  let isOnTrack = useRef<boolean | undefined>(false);
```
Comment (snowystinger) [inline packages/@react-aria/color/src/useColorWheel.ts:119]: same here, i see no reason for it to be undefined

File: packages/@react-spectrum/progress/src/ProgressBarBase.tsx [hunk 3]
```diff
@@ -86,7 +86,7 @@ function ProgressBarBase(props: ProgressBarBaseProps, ref: DOMRef<HTMLDivElement
       }
       {showValueLabel &&
         <div className={classNames(styles, 'spectrum-BarLoader-percentage')}>
-          {barProps['aria-valuetext']}
+          {barProps?.['aria-valuetext']}
```
Comment (snowystinger) [inline packages/@react-spectrum/progress/src/ProgressBarBase.tsx:89]: I'd add a check to line 89 for barProps existing, that way we don't render an empty percentage area

**Review by snowystinger — COMMENTED** [review]
Looks pretty good so far

**Review by snowystinger — APPROVED** [review]

**Review by dannify — DISMISSED** [review]

File: packages/@react-spectrum/label/stories/HelpText.stories.tsx [hunk 4]
```diff
@@ -127,7 +127,8 @@ function TextFieldWithValidationState(props: SpectrumTextFieldProps) {
         {...props}
         value={value}
         onChange={setValue}
-        validationState={value.length ? 'invalid' : valid} />
+        // eslint-disable-next-line
+        validationState={value.length ? 'invalid' : (valid === 'valid' ? 'valid' : undefined)} />
```
[the remark this answers is not in the document]
Comment (yihuiliao) [reply packages/@react-spectrum/label/stories/HelpText.stories.tsx:131]: it was for the nested ternary but i've updated the code so it doesn't use one

**Review by snowystinger — APPROVED** [review]

**Review by dannify — APPROVED** [review]

**Review by iamwillpowell — APPROVED** [review]

File: packages/@react-spectrum/form/stories/Form.stories.tsx [hunk 5]
```diff
@@ -604,12 +605,14 @@ function FormWithSubmit() {
 
   return (
     <Form onSubmit={handleSubmit} isReadOnly={formStatus === 'valid'}>
-      {(formStatus === 'invalid' || formStatus === 'valid') &&
-        <InlineAlert variant={formStatus === 'invalid' ? 'negative' : 'positive'}>
-          <Header>{formStatus === 'invalid' ? 'Error' : 'Success'}</Header>
-          <Content>{formStatus === 'invalid' ? 'There was an error with the form.' : 'Form was successfully completed.'}</Content>
-        </InlineAlert>
-      }
+      <>
```
Comment (yihuiliao) [inline packages/@react-spectrum/form/stories/Form.stories.tsx:608]: as for why this is wrapped in a empty react fragment, see [stackoverflow comment](https://stackoverflow.com/questions/62612062/ts2322-false-element-is-not-assignable-to-type-reactelement-error-does-not)

**Pull request merged**